monai/data/image_writer.py

```diff
@@ -0,0 +1,395 @@
+            return data, ensure_mat44(target_affine)
+
+        # resolve orientation
+        if has_nib:  # this is to avoid dependency on nibabel
+            data, _affine = reorient_spatial_axes(data, affine, target_affine)
+            if np.allclose(_affine, target_affine, atol=AFFINE_TOL):
+                return data, ensure_mat44(_affine)
+
+        # need resampling
+        dtype = dtype or data.dtype  # type: ignore
+        if output_spatial_shape is None:
+            output_spatial_shape, _ = compute_shape_offset(data.shape, _affine, target_affine)
+        output_spatial_shape_ = list(output_spatial_shape) if output_spatial_shape is not None else []
+        sp_dims = min(data.ndim, 3)
+        output_spatial_shape_ += [1] * (sp_dims - len(output_spatial_shape_))
+        output_spatial_shape_ = output_spatial_shape_[:sp_dims]
+        original_channels = data.shape[3:]
+        if original_channels:  # multi channel, resampling each channel
+            data_np: np.ndarray = data.reshape(list(data.shape[:3]) + [-1])  # type: ignore
+            data_np = np.moveaxis(data_np, -1, 0)  # channel first for pytorch
+        else:  # single channel image, need to expand to have a channel
+            data_np = data[None]
+        affine_xform = AffineTransform(
+            normalized=False, mode=mode, padding_mode=padding_mode, align_corners=align_corners, reverse_indexing=True
+        )
+        data_torch = affine_xform(
+            torch.as_tensor(np.ascontiguousarray(data_np, dtype=dtype)).unsqueeze(0),
+            torch.as_tensor(np.ascontiguousarray(np.linalg.inv(_affine) @ target_affine, dtype=dtype)),
+            spatial_size=output_spatial_shape_,
+        )
+        data_np = data_torch[0].detach().cpu().numpy()
+        if original_channels:
+            data_np = np.moveaxis(data_np, 0, -1)  # channel last
+            data_np = data_np.reshape(list(data_np.shape[:3]) + list(original_channels))
+        else:
+            data_np = data_np[0]
+        return data_np, ensure_mat44(target_affine)
+
+    @classmethod
+    def convert_to_channel_last(
+        cls,
+        data: NdarrayOrTensor,
+        channel_dim: Optional[int] = 0,
+        squeeze_end_dims: bool = True,
+        spatial_ndim: Optional[int] = 3,
+        contiguous: bool = False,
+    ):
+        """
+        Rearrange the data array axes to make the `channel_dim`-th dim the last
+        dimension and ensure there are three spatial dimensions. If
+        ``channel_dim`` is ``None``, a new axis will be appended to the last
+        dimension.
+
+        When ``squeeze_end_dims`` is ``True``, a postprocessing step will be
+        applied to remove any trailing singleton dimensions.
+
+        Args:
+            data: input data to be converted to "channel-last" format.
+            channel_dim: specifies the axis of the data array that is the channel dimension.
+                ``None`` indicates no channel dimension.
```

Comment:
Could you please help describe the supported values? I remember it can be multiple channels dims now?
I checked below code, seems it only supports 1 channel dim: `while len(data.shape) < spatial_ndim + 1`.

Thanks.